File: src/main/java/com/source/plusutil/service/noticeService/NoticeService.java
package com.source.plusutil.service.noticeService;

import javax.servlet.http.HttpServletRequest;

import org.springframework.security.core.Authentication;

public interface NoticeService {

	public void getNoticeList(HttpServletRequest request, Authentication authentication, Integer currentPage);
	
	public void writeNotice(String noticeTitle, String noticeContent, String category, HttpServletRequest request, Authentication authentication);
	
	public Integer getNoticeTotalPage(Integer currentPage, Integer listSize);
	
	public void getNoticeDetailInfo(HttpServletRequest request, Authentication authentication, Integer noticeNo);
	
	public void deleteNoticeInfo(HttpServletRequest request, Authentication authentication, Integer noticeNo);
	
	public void updateNoticeInfo(HttpServletRequest request, Authentication authentication, Integer noticeNo , String noticeTitle, String noticeContent);
}

File: src/main/java/com/source/plusutil/service/noticeService/NoticeService.kt
package com.source.plusutil.service.noticeService

import com.source.plusutil.dto.notice.*
import org.springframework.security.core.Authentication
import javax.servlet.http.HttpServletRequest

interface NoticeService {
    fun getNoticeList(authentication: Authentication?, currentPage: Int?): NoticeListDto?
    fun writeNotice(noticeWriteDto: NoticeWriteRequestDto, authentication: Authentication?) : NoticeWriteResponseDto
    fun getNoticeTotalPage(currentPage: Int?, listSize: Int?): Int?
    fun getNoticeDetailInfo(authentication: Authentication?, noticeNo: Long?): NoticeDetailDto?
    fun deleteNoticeInfo(noticeNo: Long?, currentPage: Int?) : NoticeDeleteResponseDto?
    fun updateNoticeInfo(request: HttpServletRequest?, authentication: Authentication?, noticeNo: Long?, noticeTitle: String?, noticeContent: String?, category: String?)
}